java/DataLake/src/main/java/Reader.java

```diff
@@ -75,8 +78,20 @@ public Book readBook(String path) {
         } catch (IOException e) {
             e.printStackTrace();
         }
+        bookName = this.extractBookName(fileName);
+        Book book = new Book(bookName,words);
+        datalake.addBook(book.getIndex(), book.getName());
+
+        return book;
+    }
+
+    public static String extractBookName(String fileName) {
+        String[] parts = fileName.split("\\)");
+        String extractedName = parts[1].replace("_", " ");
```

Comment:
The app sometimes shuts down on this step. Don't know how to replicate it.

Reply:
i think it should work fine if we remove the regular expression and just put ")" inside the split method @adam-prochazka

Reply:
ok so nope there is no other way of doing it

Reply:
We assume that the fileName follows the Crawler downloaded books structure thing like:

(GUTENBER ID) book name.txt 

it wont work if the file is not like this.